### PR TITLE
Minor refactoring of DPW so the code structure is more consistent

### PR DESCRIPTION
**Pull request by kykim0**
More consistent with the vanilla MCTS code, that is.

In particular, moved out the action selection logic to separate methods.

### CODE REVIEW

File: src/dpw_types.jl [hunk 1]
```diff
@@ -142,13 +142,13 @@ function DPWSolver(;depth::Int=10,
...
                     tree_in_info::Bool=false,
                     rng::AbstractRNG=Random.GLOBAL_RNG,
-                    estimate_value::Any = RolloutEstimator(RandomSolver(rng)),
-                    init_Q::Any = 0.0,
-                    init_N::Any = 0,
-                    next_action::Any = RandomActionGenerator(rng),
-                    default_action::Any = ExceptionRethrow(),
-                    reset_callback::Function = (mdp, s)->false,
-                    show_progress::Bool = false,
+                    estimate_value::Any=RolloutEstimator(RandomSolver(rng)),
```
Comment (zsunberg) [inline src/dpw_types.jl:145]: In the future, please try to avoid small formatting changes unrelated to the PR, it makes it harder to review.

File: src/dpw.jl [hunk 2]
```diff
@@ -96,7 +83,7 @@ function simulate(dpw::DPWPlanner, snode::Int, d::Int)
     dpw.reset_callback(dpw.mdp, s) # Optional: used to reset/reinitialize MDP to a given state.
     if isterminal(dpw.mdp, s)
         return 0.0
-    elseif d == 0
+    elseif d == 0 || (timeout_us > 0.0 && CPUtime_us() > timeout_us)
```
Comment (zsunberg) [inline src/dpw.jl]: This seems redundant with the timeout check in the search loop and unrelated to the main PR subject. Why was it added?

File: src/dpw.jl [hunk 3]
```diff
@@ -28,13 +28,8 @@ function POMDPModelTools.action_info(p::DPWPlanner, s; tree_in_info=false)
...
         S = statetype(p.mdp)
         A = actiontype(p.mdp)
-        if p.solver.keep_tree
-            if p.tree == nothing
-                tree = DPWTree{S,A}(p.solver.n_iterations)
-                p.tree = tree
-            else
-                tree = p.tree
-            end
+        if p.solver.keep_tree && p.tree != nothing
```
Comment (zsunberg) [inline src/dpw.jl:31]: This change seems like it is probably OK so we can keep it this time, but changes that are not related to the main subject of the PR slow down reviews, so they should usually be avoided or submitted in other PRs

**Review by zsunberg — CHANGES_REQUESTED** [review]
Thanks for the contribution!

The main change of moving the node selection into functions looks good. However, there are a lot of other changes not related to this in the PR.

In particular, please remove the additional time check before we merge this since we haven't determined whether it is necessary.

(I realize that it is a bit of a pain to do this, but in the long run, simple, atomic, pull requests that address one issue will make maintenance much better!)

File: src/dpw_types.jl (hunk 1 above)
Comment (kykim0) [reply src/dpw_types.jl:145]: Will do, thanks!

File: src/dpw.jl (hunk 2 above)
Comment (kykim0) [reply src/dpw.jl]: Thanks for catching this.  This has been removed.  It was needed for the version of parallel MCTS implementation I was working and crept into this PR.

File: src/dpw.jl (hunk 3 above)
Comment (kykim0) [reply src/dpw.jl:31]: Thanks, couldn't resist packing small changes into the same PR but will avoid it in future PRs :-)

**Comment by zsunberg** [comment]
Awesome! Thanks. This looks good.

**Comment by kykim0** [comment]
Thanks a lot for the quick reviews!
